backend/compact-connect/lambdas/python/common/tests/unit/test_provider_record_util.py

```diff
@@ -0,0 +1,137 @@
+        self.assertEqual(best_license['compactEligibility'], CompactEligibilityStatus.ELIGIBLE)
+
+    def test_find_best_license_active_preferred_when_no_compact_eligible_licenses(self):
+        """Test that find_best_license prefers active licenses when no compact-eligible licenses exist."""
+        from cc_common.data_model.provider_record_util import ProviderRecordUtility
+        from cc_common.data_model.schema.common import ActiveInactiveStatus, CompactEligibilityStatus
+
+        licenses = [
+            {
+                **self.base_license,
+                'dateOfIssuance': '2024-01-01',
+                'licenseStatus': ActiveInactiveStatus.ACTIVE,
+                'compactEligibility': CompactEligibilityStatus.INELIGIBLE,
+            },
+            {
+                **self.base_license,
+                'dateOfIssuance': '2024-02-01',
+                'licenseStatus': ActiveInactiveStatus.INACTIVE,
+                'compactEligibility': CompactEligibilityStatus.INELIGIBLE,
+            },
+        ]
+
+        best_license = ProviderRecordUtility.find_best_license(licenses)
+        self.assertEqual(best_license['dateOfIssuance'], '2024-01-01')
+        self.assertEqual(best_license['licenseStatus'], ActiveInactiveStatus.ACTIVE)
+
+    def test_find_best_license_most_recent_when_no_active_or_compact_eligible_licenses(self):
+        """Test that find_best_license selects most recent license when no active or compact-eligible licenses exist."""
+        from cc_common.data_model.provider_record_util import ProviderRecordUtility
+        from cc_common.data_model.schema.common import ActiveInactiveStatus, CompactEligibilityStatus
+
+        licenses = [
+            {
+                **self.base_license,
+                'dateOfIssuance': '2024-01-01',
+                'licenseStatus': ActiveInactiveStatus.INACTIVE,
+                'compactEligibility': CompactEligibilityStatus.INELIGIBLE,
+            },
+            {
+                **self.base_license,
+                'dateOfIssuance': '2024-02-01',
+                'licenseStatus': ActiveInactiveStatus.INACTIVE,
+                'compactEligibility': CompactEligibilityStatus.INELIGIBLE,
+            },
+        ]
+
+        best_license = ProviderRecordUtility.find_best_license(licenses)
+        self.assertEqual(best_license['dateOfIssuance'], '2024-02-01')
+
+    def test_find_best_license_raises_exception_when_no_licenses(self):
+        """Test that find_best_license raises an exception when no licenses are provided."""
+        from cc_common.data_model.provider_record_util import ProviderRecordUtility
+        from cc_common.exceptions import CCInternalException
+
+        with self.assertRaises(CCInternalException):
+            ProviderRecordUtility.find_best_license([])
+
+    def test_find_best_license_complex_scenario(self):
+        """Test a complex scenario with multiple licenses of different types and statuses."""
```

Comment:
nit but this seems to be only testing different statues, not types–

```suggestion
        """Test a complex scenario with multiple licenses of different statuses."""
```